Lib/importlib/_bootstrap_external.py

```diff
@@ -802,7 +802,7 @@ def __init__(self, fullname, path):
         """Cache the module name and the path to the file found by the
         finder."""
         self.name = fullname
-        self.path = path
+        self.path = path.__fspath__() if hasattr(path, '__fspath__') else path
```

Comment:
@brettcannon I think we will still need to use this trick, the test cases including `None`, `list` and others , if we pass it to `_os.fspath`, it will raise the `TypeError`.